src/environments/array.js

```diff
@@ -30,8 +30,22 @@ export type ArrayEnvNodeData = {
     // initialization.
     body?: ParseNode<*>[][], // List of rows in the (2D) array.
     rowGaps?: number[],
+    numHLinesBeforeRow?: number[],
 };
 
+function nextTokenIsHLine(parser: Parser): boolean {
+    let next = parser.nextToken.text;
+    while (/^\s$/.test(next)) {
+        parser.consume();
+        next = parser.nextToken.text;
+    }
```

Comment:
Could this be done using `parser.consumeSpaces()`?

Reply:
Yes. Please use that instead.

Reply:
Will do.